protocols/gossipsub/src/config.rs

```diff
@@ -133,9 +134,11 @@ pub struct GossipsubConfig {
     /// addressing, where this function may be set to `hash(message)`. This would prevent messages
     /// of the same content from being duplicated.
     ///
-    /// The function takes a `GossipsubMessage` as input and outputs a String to be interpreted as
-    /// the message id.
-    message_id_fn: fn(&GossipsubMessage) -> MessageId,
+    /// The function takes a `GenericGossipsubMessage` as input and outputs a String to be
+    /// interpreted as the message id.
+    message_id_fn: fn(&GenericGossipsubMessage<T>) -> MessageId,
+
+    fast_message_id_fn: Option<fn(&RawGossipsubMessage) -> MessageId>,
```

Comment:
Can we add some description to this guy

Reply:
Oh forgot that, added a description now.